lightning/src/ln/channelmanager.rs

```diff
@@ -207,6 +207,12 @@ pub(super) enum HTLCFailReason {
 	}
 }
 
+/// Return value for claim_funds_from_hop
+struct ClaimFundsRes {
```

Comment:
Could we switch this to be an enum, something along these lines:
```
enum ClaimFundsRes {
	PrevHopForceClosed,
	MonitorUpdateFailed((u64, PublicKey, MsgHandleErrInternal)),
... other states
}
```

This refactor is @jkczyz -approved :stuck_out_tongue:

Reply:
Hmm, I'm not sure that that's that much clearer. All four possible states of `(None, Ok(..))`, `(Some(..), Ok(..))`, `(None, Err(..))`, and `(None, Ok(..))` are used, so we'd need at least four variants, and we'd have a really hard time keeping the callsite `match` DRY.

Reply:
IMO seeing a descriptive name like `PrevHopForceClosed` makes the code more readable at a glance than seeing `(None, Ok(..))`.

It seems like the callsite might only need to repeat a few lines? Is that off?

Reply:
I mean I can split it into a struct containing two enums, but that seems to border on enough volume that it becomes unreadable just by weight. Sticking it all in one enum (with four variants) would be really harsh for `claim-funds`, too, which only actually cares about one of the two fields, now it would have to have a match and conversion instead of being able to just access the field. I agree having names for things is useful, but I don't think I agree it is if it results in more code by volume including matches that the reader has to figure out. Maybe there's a way to rename the fields in `ClaimFundsRes` to make it more clear instead?

Reply:
Briefly jammed with @jkczyz on this patch: https://github.com/valentinewallace/rust-lightning/commit/f9d338daa944b910ad9178421c916f6ce6ebdd64 It seems more readable and the code volume doesn't seem significantly increased!

Reply:
Yea, I guess I underestimated the amount of parameter conversion in the current code. I do think the `MonitorUpdateFail` overload that can mean *both* "channel is bogus, closing" and `Success`, we claimed is a bit confusing, but its not strictly worse than it was. I took that commit and added one tweak.